sklearn/cross_validation.py

```diff
@@ -259,6 +259,11 @@ def __init__(self, n, n_folds, indices, shuffle, random_state):
         if not isinstance(shuffle, bool):
             raise TypeError("shuffle must be True or False;"
                             " got {0}".format(shuffle))
+
+        if random_state is not None and not shuffle:
+            warnings.warn("`shuffle` should be set to True for the"
+                          " data to be shuffled based on the given `random_state`.")
```

Comment:
You could also refactor this to a function `check_random_state_shuffle(random_state, shuffle)` and move it to `sklearn/utils/validation.py` and use this function here :)